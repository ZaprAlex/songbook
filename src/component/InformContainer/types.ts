import { IButton } from '../Button';

export interface IInformContainer {
    icon?: string;
    header: string;
    headerClassName?: string;
    message?: string;
    messageClassName?: string;
    buttonLabel?: string;
    buttonCancelLabel?: string;
    contentClassName?: string;
    buttonClassName?: string;
    buttonCancelClassName?: string;
    onMount?: () => void;
    onClick?: () => void;
    onCancelClick?: () => void;
    buttonCancelProps?: IButton;
    buttonProps?: IButton;
    loading?: boolean;
}
