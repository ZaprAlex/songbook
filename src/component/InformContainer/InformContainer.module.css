.content {
    width: 100%;
    display: flex;
    flex-flow: column nowrap;
    align-items: center;
}

.icon-wrapper {
    width: 100%;
    height: 256px;
    display: flex;
    flex-flow: column nowrap;
    align-items: center;
}

.icon {
    height: 100%;
}

.header {
    width: 100%;
    margin-top: 8px;
    font-size: 32px;
    line-height: 1.25;
    letter-spacing: -0.3px;
    text-align: center;
    white-space: pre-wrap;
    color: #000000;
}

.message {
    width: 100%;
    margin-top: 16px;
    font-size: 18px;
    line-height: 1.56;
    letter-spacing: -0.3px;
    text-align: center;
    white-space: pre-wrap;
    color: rgba(0, 0, 0, 0.4);
}

.button-block {
    width: 100%;
    display: flex;
    flex-flow: row nowrap;
    justify-content: center;
    align-items: center;
}

.button-block > *:not(:first-child) {
    margin-left: 24px;
}
