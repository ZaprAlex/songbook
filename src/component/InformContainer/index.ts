export { default } from './InformContainer';
export * from './types';
