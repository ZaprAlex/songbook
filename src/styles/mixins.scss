@import './variables';

@mixin adaptive-for($device) {
    @if $device == mobile {
        @media (max-width: $min-tablet-width - 1) {
            @content;
        }
    } @else if $device == tablet {
        @media (min-width: $min-tablet-width) and (max-width: $min-desktop-width - 1) {
            @content;
        }
    } @else if $device == desktop {
        @media (min-width: $min-desktop-width) {
            @content;
        }
    }
}

@mixin adaptive-for-mobile {
    @include adaptive-for(mobile) {
        @content;
    }
}

@mixin adaptive-for-desktop {
    @include adaptive-for(desktop) {
        @content;
    }
}

@mixin adaptive-except-mobile {
    @media (min-width: $min-tablet-width) {
        @content;
    }
}
