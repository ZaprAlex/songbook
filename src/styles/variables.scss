$min-tablet-width: 576px;
$min-desktop-width: 992px;

$max-content-width: 1224px;
$max-outer-content-width: calc(#{$max-content-width} + 280px);
