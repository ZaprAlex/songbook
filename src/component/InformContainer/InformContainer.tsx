import React, { useEffect } from 'react';
import cn from 'classnames';

import { noop } from '../../utils/helper';
import Button from '../Button';

import { IInformContainer } from './types';

import styles from './InformContainer.module.css';

const InformContainer: React.FC<IInformContainer> = ({
    icon,
    header,
    headerClassName,
    message,
    messageClassName,
    buttonLabel,
    buttonCancelLabel,
    onMount = noop,
    onClick,
    onCancelClick,
    contentClassName,
    buttonClassName,
    buttonCancelClassName,
    buttonCancelProps,
    buttonProps,
    loading,
}) => {
    useEffect(onMount, []);

    return (
        <div className={cn(styles.content, contentClassName)}>
            <div className={styles.iconWrapper}>
                <img src={icon} alt="" className={styles.icon} data-test-id="inform-container-icon" />
            </div>
            <div className={cn(styles.header, headerClassName)}>{header}</div>
            {message && <div className={cn(styles.message, messageClassName)}>{message}</div>}
            <div className={styles.buttonBlock}>
                {onCancelClick && buttonCancelLabel && (
                    <Button
                        {...buttonCancelProps}
                        type="outline"
                        label={buttonCancelLabel}
                        onClick={onCancelClick}
                        className={buttonCancelClassName ?? buttonClassName}
                        dataTestId="button-cancel"
                        loading={loading}
                    />
                )}
                {onClick && buttonLabel && (
                    <Button
                        {...buttonProps}
                        label={buttonLabel}
                        type="contained"
                        skeuomorph
                        onClick={onClick}
                        className={buttonClassName}
                        dataTestId="button-submit"
                        loading={loading}
                    />
                )}
            </div>
        </div>
    );
};

export default InformContainer;
