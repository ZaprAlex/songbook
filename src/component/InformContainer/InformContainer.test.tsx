import React from 'react';
import { shallow } from 'enzyme';

import icon from './../../static/svgs/not-found.svg';

import InformContainer, { IInformContainer } from './index';
import styles from './InformContainer.module.css';

const testHeader = 'TestHeader';
const testButtonLabel = 'TestButtonLabel';
const testMessage = 'TestMessage';

test('w/o message and btn', () => {
    const initialProps: IInformContainer = {
        header: testHeader,
        icon: icon,
        contentClassName: styles.content,
    };
    const body = shallow(<InformContainer {...initialProps} />);
    expect(body.find('.header').text()).toEqual(testHeader);
    expect(body.html()).toMatchSnapshot();
});

test('full functionality', () => {
    const buttonCallBack = jest.fn();
    const initialProps: IInformContainer = {
        header: testHeader,
        message: testMessage,
        icon: icon,
        contentClassName: styles.content,
        buttonLabel: testButtonLabel,
        onClick: buttonCallBack,
        buttonClassName: styles.button,
    };
    const body = shallow(<InformContainer {...initialProps} />);
    expect(body.find('.header').text()).toEqual(testHeader);
    expect(body.find('.message').text()).toEqual(testMessage);
    const submitButton = body.find('.button');
    expect(submitButton.props().label).toEqual(testButtonLabel);
    expect(body.html()).toMatchSnapshot();
    submitButton.simulate('click');
    expect(buttonCallBack.mock.calls.length).toEqual(1);
});
