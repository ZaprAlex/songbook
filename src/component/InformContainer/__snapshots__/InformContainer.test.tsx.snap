// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`full functionality 1`] = `"<div class=\\"content content\\"><div class=\\"header\\">TestHeader</div><div class=\\"message\\">TestMessage</div><div class=\\"buttonBlock\\"><button type=\\"button\\" class=\\"btn btn-green btn-contained btn-middle btnSkeuomorph button\\" data-test-id=\\"button-submit\\"><span class=\\"buttonLabel\\">TestButtonLabel</span></button></div></div>"`;

exports[`w/o message and btn 1`] = `"<div class=\\"content content\\"><div class=\\"header\\">TestHeader</div><div class=\\"buttonBlock\\"></div></div>"`;
